src/main/grammars/WebAssemblyLexer.flex

```diff
@@ -20,30 +20,123 @@ import org.jetbrains.webstorm.lang.psi.WebAssemblyTypes;
+                                      | rot[lr] | eqz? | ne | [lg][te] _[su])
+             | f (32 | 64) \. (abs | neg | ceil | floor | trunc | nearest | sqrt | add | sub | mul | div | min | max
+                                   | copysign | eq | ne | [lg][te])
+             | i32 \. wrap_i64
+             | i (32 | 64) \. trunc_ ((f | sat) (32 | 64) _ [su])
+             | i64 \. extend_i32[su]
+             | f (32 | 64) \. convert_ (i (32 | 64) _ [su])
+             | f32 \. demote_f64 | f64 \. promote_f32
+             | i32 \. reinterpret_f32 | i64 \. reinterpret_f64 | f32 \. reinterpret_i32 | f64 \. reinterpret_i64
+             | i32 \. extend (8 | 16) _s | i64 \. extend (8 | 16 | 32) _s
+VARIABLEINSTR = ("local" | "global") \. ("get" | "set") | "local.tee"
+MEMORYINSTR = {VALTYPE} \. ("load" | "store")
+            | "i32.load"  (8 | 16)_[su] | "i64.load"  (8 | 16 | 32)_[su]
+            | "i32.store" (8 | 16) | "i64.store" (8 | 16 | 32)
 
 %%
+<YYINITIAL> {
+    {VALTYPE}                                       { return WebAssemblyTypes.VALTYPE; }
+    {ONEWORDINSTR}                                  { return WebAssemblyTypes.ONEWORDINSTR; }
+    {VARIABLEINSTR}                                 { return WebAssemblyTypes.VARIABLEINSTR; }
+    {MEMORYINSTR}                                   { return WebAssemblyTypes.MEMORYINSTR; }
+
+    "func"                                          { return WebAssemblyTypes.FUNCKEY; }
+    "param"                                         { return WebAssemblyTypes.PARAMKEY; }
+    "result"                                        { return WebAssemblyTypes.RESULTKEY; }
+    "funcref"                                       { return WebAssemblyTypes.FUNCREFKEY; }
+    "mut"                                           { return WebAssemblyTypes.MUTKEY; }
 
-<YYINITIAL> {END_OF_LINE_COMMENT}                           { yybegin(YYINITIAL); return WebAssemblyTypes.COMMENT; }
+    "block"                                         { return WebAssemblyTypes.BLOCKKEY; }
+    "loop"                                          { return WebAssemblyTypes.LOOPKEY; }
+    "end"                                           { return WebAssemblyTypes.ENDKEY; }
+    "if"                                            { return WebAssemblyTypes.IFKEY; }
+    "else"                                          { return WebAssemblyTypes.ELSEKEY; }
+    "br"                                            { return WebAssemblyTypes.BRKEY; }
+    "br_if"                                         { return WebAssemblyTypes.BRIFKEY; }
+    "call"                                          { return WebAssemblyTypes.CALLKEY; }
+    "br_table"                                      { return WebAssemblyTypes.BRTABLEKEY; }
+    "call_indirect"                                 { return WebAssemblyTypes.CALLINDIRECTKEY; }
+    "offset="                                       { return WebAssemblyTypes.OFFSETEQKEY; }
```

Comment:
are you sure that it is a token? e.g. that after "offset" we cannot have block comments or spaces?

Reply:
Yes, I'm sure:
https://webassembly.github.io/spec/core/text/instructions.html#memory-instructions